File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    var answer = 0;
    let res = new Set();
    const n = elements.length
    for (let i = 1; i<n+1; i++){
        for (let st = 0; st<n; st++){
            let cnt = 0
            let tot = 0
            for (let idx = st; cnt < i; idx++){
                tot += elements[idx%n]
                cnt += 1
            }
            res.add(tot)
        }
    }
    answer = res.size
    return answer;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        HashSet<Integer> results = new HashSet<>();
        for (int k: elements) {
            results.add(k);
        }
        
        for (int l=2; l<elements.length+1; l++) {
            int initialSum = 0;
            for (int i=0; i<l; i++) {
                initialSum += elements[i];
            }
            results.add(initialSum);
            
            for (int st=1; st<elements.length; st++) {
                initialSum -= elements[st-1];
                int en = (st + l - 1) % elements.length;
                initialSum += elements[en];
                results.add(initialSum);
            }
        }
        
        answer = results.size();
        return answer;
    }
}